src/managed/OpenLiveWriter.BlogClient/Clients/WordPressClient.cs

```diff
@@ -53,5 +54,49 @@ protected override void ConfigureClientOptions(BlogClientOptions clientOptions)
             clientOptions.SupportsAuthor = true;
         }
 
+        public override BlogInfo[] GetUsersBlogs()
+        {
+            TransientCredentials tc = Login();
+            return GetWPUsersBlogs(tc.Username, tc.Password);
+        }
+        private BlogInfo[] GetWPUsersBlogs(string username, string password)
+        {
+            // call method
+            XmlNode result = CallMethod("wp.getUsersBlogs",
+                new XmlRpcString(username),
+                new XmlRpcString(password, true));
+
+            try
+            {
+                // parse results
+                ArrayList blogs = new ArrayList();
+                XmlNodeList dataValues = result.SelectNodes("array/data");
+                foreach (XmlNode dataValue in dataValues)
+                {
+                    XmlNodeList blogNodes = dataValue.SelectNodes("value/struct");
+                    foreach (XmlNode blogNode in blogNodes)
+                    {
+                        // get node values
+                        XmlNode idNode = blogNode.SelectSingleNode("member[name='blogid']/value");
+                        XmlNode nameNode = blogNode.SelectSingleNode("member[name='blogName']/value");
+                        XmlNode urlNode = blogNode.SelectSingleNode("member[name='url']/value");
```

Comment:
Confirmed that the correct endpoint gets set, but not because the function returns multiple blogs. The function is only called during a ValidateService() function call. The only blog that gets added is the one with the correct sub-domain compared to the URL entered at the beginning of the adding of the blog account. This surprises me, but doesn't open up new issues.